Hacklift/communual-api
# Title: Email confirmation upon successful registration

### What does this PR do?
- This PR implement the functionality that sends a confirmation email to a user upon successful registration so that the user can confirm the email address inputted upon registration

### Description of task to be completed:
- Configure and set up sendgrid in application.
- Implement a method that handles  the function that sends email to a user upon successful registration.
- Implement a method that handles the function that sends another email to a user on request by the user so as to re-verify user email address if the previous token has expired.
- Implement endpoints for both features
- Implement test

### How can this be manually tested?
- Run server on command `python run.py`
- Register as a user with a valid email address. `localhost:5000/api/auth/signup`
- Visit email address and click on the email confirmation link to confirm your email before 1 hour passes. You get directed to `http://localhost:5000/api/auth/email-confirmation/?emailtoken=*****`
- Hit endpoint on postman to test for email re-verification `localhost:5000/api/auth/reverify-email`

![email_confirmation](https://user-images.githubusercontent.com/32917511/52168028-cd689580-2724-11e9-9d3c-66d2bc5fc06b.PNG)


### Link to Trello Board Card:
[Card_id: reP0Ohci](https://trello.com/c/reP0Ohci/19-as-a-user-i-should-be-able-to-receive-a-mail-to-confirm-user-email-address-upon-registration)
